src/tasks.h

```diff
@@ -41,17 +45,17 @@ class Task
 	TaskFunc func;
 };
 
-Task* createTask(TaskFunc&& f);
-Task* createTask(uint32_t expiration, TaskFunc&& f);
+Task_ptr createTask(TaskFunc&& f);
+Task_ptr createTask(uint32_t expiration, TaskFunc&& f);
 
 class Dispatcher : public ThreadHolder<Dispatcher>
 {
 public:
-	void addTask(Task* task);
+	void addTask(Task_ptr task);
 
-	void addTask(TaskFunc&& f) { addTask(new Task(std::move(f))); }
+	void addTask(TaskFunc&& f) { addTask(std::make_unique<Task>(std::move(f))); }
```

Comment:
```suggestion
	void addTask(TaskFunc&& f) { addTask(std::make_unique<Task>(std::forward(f))); }
```

Reply:
Can you explain why std::forward?

Reply:
> Can you explain why std::forward?

To maintain the current value type (L or R) for the meta side of things